commit 9a4328600c9c8ce65e37515694b105454946c20b
Author: Omar Einea <eineao@gmail.com>
Date:   Sun Apr 2 11:21:25 2017 +0300

    Deploying Project On Heroku

diff --git a/.gitignore b/.gitignore
index 44f3ad3..b5ef717 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,8 +1,5 @@
-# Ignore Interpreted Python Files
+venv
 *.pyc
-
-# Ignore PyCharm Project folder
+staticfiles
+.env
 .idea
-
-# Ignore Django Static Files folder
-static
diff --git a/Procfile b/Procfile
new file mode 100644
index 0000000..b9c12a6
--- /dev/null
+++ b/Procfile
@@ -0,0 +1 @@
+web: gunicorn UOSHUB.wsgi
diff --git a/UOSHUB/settings.py b/UOSHUB/settings.py
index 03da1ba..ddc62f7 100644
--- a/UOSHUB/settings.py
+++ b/UOSHUB/settings.py
@@ -12,9 +12,11 @@ https://docs.djangoproject.com/en/1.11/ref/settings/

 import os
 import netifaces
+import dj_database_url

 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+PROJECT_ROOT = os.path.dirname(os.path.abspath(__file__))


 # Quick-start development settings - unsuitable for production
@@ -55,6 +57,7 @@ INSTALLED_APPS = [
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
+    'whitenoise.runserver_nostatic',
 ]

 MIDDLEWARE = [
@@ -132,10 +135,27 @@ USE_L10N = True
 USE_TZ = True


+# Update database configuration with $DATABASE_URL.
+db_from_env = dj_database_url.config(conn_max_age=500)
+DATABASES['default'].update(db_from_env)
+
+# Honor the 'X-Forwarded-Proto' header for request.is_secure()
+SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+
+# Allow all host headers
+ALLOWED_HOSTS = ['*']
+
 # Static files (CSS, JavaScript, Images)
 # https://docs.djangoproject.com/en/1.11/howto/static-files/

+STATIC_ROOT = os.path.join(PROJECT_ROOT, 'staticfiles')
 STATIC_URL = '/static/'

-# This is necessary so that Nginx can handle requests for static items
-STATIC_ROOT = os.path.join(BASE_DIR, 'static/')
+# Extra places for collectstatic to find static files.
+STATICFILES_DIRS = [
+    os.path.join(PROJECT_ROOT, 'static'),
+]
+
+# Simplified static file serving.
+# https://warehouse.python.org/project/whitenoise/
+STATICFILES_STORAGE = 'whitenoise.storage.CompressedManifestStaticFilesStorage'
diff --git a/UOSHUB/wsgi.py b/UOSHUB/wsgi.py
index c4dcca7..2558442 100644
--- a/UOSHUB/wsgi.py
+++ b/UOSHUB/wsgi.py
@@ -10,7 +10,9 @@ https://docs.djangoproject.com/en/1.11/howto/deployment/wsgi/
 import os

 from django.core.wsgi import get_wsgi_application
+from whitenoise.django import DjangoWhiteNoise

 os.environ.setdefault("DJANGO_SETTINGS_MODULE", "UOSHUB.settings")

 application = get_wsgi_application()
+application = DjangoWhiteNoise(application)
diff --git a/requirements.txt b/requirements.txt
index 561b16d..bd3d5e2 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -4,3 +4,7 @@ html5lib==0.999999999
 lxml==3.7.3
 mechanize==0.3.1
 netifaces==0.10.5
+whitenoise==3.2
+dj-database-url==0.4.1
+gunicorn==19.6.0
+psycopg2==2.6.2
diff --git a/runtime.txt b/runtime.txt
new file mode 100644
index 0000000..4b38fc9
--- /dev/null
+++ b/runtime.txt
@@ -0,0 +1 @@
+python-2.7.13
